lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -7,25 +7,21 @@ define((require, exports, module) ->
         placement: "bottom"
         trigger: "manual"
         title: "Formula"
-        container: "#editor"
+        container: editor.container
       }
 
-      show: (jqPopoverContainer, content) ->
-        cursorPosition = $("textarea.ace_text-input").position()
-        jqPopoverContainer.css({
-          top: "#{cursorPosition.top + 24}px"
-          left: "#{cursorPosition.left}px"
-        })
+      show: (jqPopoverContainer, content, position) ->
+        jqPopoverContainer.css(position)
         popoverHandler.options.content = content
         jqPopoverContainer.popover(popoverHandler.options)
         jqPopoverContainer.popover("show")
         return
 
-      hide: (jqPopoverContainer) ->
+      destroy: (jqPopoverContainer) ->
         jqPopoverContainer.popover("destroy")
 
-      isVisible: (jqPopoverElement) ->
-        jqPopoverElement.children(".popover").is(":visible")
+      isVisible: (jqPopoverContainer) ->
+        jqPopoverContainer.data().popover.tip().hasClass("in")
```

Comment:
`data().popover.tip()`, as far as I understand, is generated by Bootstrap when creating the popover under this element. So, as long as Bootstrap doesn't change its API, we should be good in this part. As for the `.hasClass("in")`, would it help to change this to `.is(":visible")`? So the whole thing would be `jqPopoverContainer.data().popover.tip().is(":visible")`. 

About possible new version of Bootstrap -- we're already working here with Bootstrap 2, which is obsolete. In Bootstrap 3, the API for a lot of things is slightly different. So a lot of code will break with the new version of Bootstrap, not just this one.